runelite-client/src/main/java/net/runelite/client/plugins/jewelleryenchanting/JewelleryEnchantingOverlay.java

```diff
@@ -0,0 +1,167 @@
+	
+	private Map<String, int[]> unenchantedItems = new HashMap<> ();
+	
+	@Inject
+	public JewelleryEnchantingOverlay (JewelleryEnchantingPlugin plugin, JewelleryEnchantingConfig config, Client client, QueryRunner queryRunner)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.plugin = plugin;
+		this.config = config;
+		this.client = client;
+		this.queryRunner = queryRunner;
+		
+		unenchantedItems.put("Lvl-1 Enchant", new int[]
+		{
+			ItemID.SAPPHIRE_RING,
+			ItemID.SAPPHIRE_NECKLACE,
+			ItemID.SAPPHIRE_BRACELET,
+			ItemID.SAPPHIRE_BRACELET_11072,
+			ItemID.SAPPHIRE_AMULET,
+			ItemID.OPAL_RING,
+			ItemID.OPAL_BRACELET,
+			ItemID.OPAL_NECKLACE,
+			ItemID.OPAL_AMULET
+		});
+		unenchantedItems.put("Lvl-2 Enchant", new int[]
+		{
+			ItemID.EMERALD_RING,
+			ItemID.EMERALD_NECKLACE,
+			ItemID.EMERALD_BRACELET,
+			ItemID.EMERALD_AMULET,
+			ItemID.JADE_RING,
+			ItemID.JADE_NECKLACE,
+			ItemID.JADE_BRACELET,
+			ItemID.JADE_AMULET
+		});
+		unenchantedItems.put("Lvl-3 Enchant", new int[]
+		{
+			ItemID.RUBY_RING,
+			ItemID.RUBY_BRACELET,
+			ItemID.RUBY_AMULET,
+			ItemID.RUBY_NECKLACE,
+			ItemID.TOPAZ_RING,
+			ItemID.TOPAZ_BRACELET,
+			ItemID.TOPAZ_NECKLACE,
+			ItemID.TOPAZ_AMULET
+		});
+		unenchantedItems.put("Lvl-4 Enchant", new int[]
+		{
+			ItemID.DIAMOND_RING,
+			ItemID.DIAMOND_NECKLACE,
+			ItemID.DIAMOND_BRACELET,
+			ItemID.DIAMOND_AMULET
+		});
+		unenchantedItems.put("Lvl-5 Enchant", new int[]
+		{
+			ItemID.DRAGONSTONE_RING,
+			ItemID.DRAGON_NECKLACE,
+			ItemID.DRAGONSTONE_BRACELET,
+			ItemID.DRAGONSTONE_AMULET
+		});
+		unenchantedItems.put("Lvl-6 Enchant", new int[]
+		{
+			ItemID.ONYX_RING,
+			ItemID.ONYX_NECKLACE,
+			ItemID.ONYX_BRACELET,
+			ItemID.ONYX_AMULET
+		});
+		unenchantedItems.put("Lvl-7 Enchant", new int[]
+		{
+			ItemID.ZENYTE_RING,
+			ItemID.ZENYTE_NECKLACE,
+			ItemID.ZENYTE_BRACELET,
+			ItemID.ZENYTE_AMULET
+		});
+	}
+	
+	@Override
+	public Dimension render (Graphics2D graphics)
+	{
+		//Make sure we only start this if the player has clicked an enchantment level
+		if (plugin.getClickedOption() == null)
+			return null;
+		
+		//Query the players inventory
+		Query inventoryQuery = new InventoryWidgetItemQuery();
+		WidgetItem[] inventoryItems = queryRunner.runQuery(inventoryQuery);
+		//Let's make sure there is items in the players inventory
+		//	Must make sure it's also not null for when the player's inventory is empty
+		if (inventoryItems != null && inventoryItems.length != 0)
+		{
+			//Let's loop through every item and display the overlay if it's an item for the current enchantment level
+			for (WidgetItem item : inventoryItems)
+			{
+				//Check to see if this item is one of the item id's in the un-enchanted list
+				if (arrayContains(unenchantedItems.get(plugin.getClickedOption()), item.getId()))
+				{
+					SpritePixels itemSprite = client.createItemSprite(item.getId(), item.getQuantity(), 1, 0, 0, true, 712);
```

Comment:
I would expand on our ItemManager to handle sprites with outlines, to avoid this which is rendering a new item sprite per item per frame ... also I would document the subtle use of `true` here preventing the client from caching the modified sprite.